rasa/core/brokers/file_producer.py

```diff
@@ -1,13 +1,14 @@
 import warnings
 
 from rasa.core.brokers.file import FileEventBroker
+from rasa.utils.common import raise_warning
 
 
 class FileProducer(FileEventBroker):
-    warnings.warn(
+    raise_warning(
         "The `FileProducer` class is deprecated, please inherit from "
         "`FileEventBroker` instead. `FileProducer` will be removed in "
         "future Rasa versions.",
-        DeprecationWarning,
-        stacklevel=2,
+        FutureWarning,
+        docs="/api/event-brokers/",
```

Comment:
optional: we could have this API sections as constants somewhere just not to update all of them every time.